src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3814,19 +3818,15 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
                 type = null;
                 name = this.EatToken(SyntaxKind.ArgListKeyword);
             }
-            SyntaxToken exclamation = null;
-            SyntaxToken equals = null;
-            if (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
-            {
-                exclamation = this.EatToken(SyntaxKind.ExclamationToken);
-            }
-            else if (this.CurrentToken.Kind == SyntaxKind.ExclamationEqualsToken)
+            // When the user type "int goo[]", give them a useful error
+            if (this.CurrentToken.Kind == SyntaxKind.OpenBracketToken && this.PeekToken(1).Kind == SyntaxKind.CloseBracketToken)
```

Comment:
This doesn't look right. It looks like we could get here by typing: `int param![]` which doesn't make sense. I would say this should be in an `else` block in the `!= SyntaxKind.ArgListKeyword` case